plugins/net_plugin/net_plugin.cpp

```diff
@@ -918,12 +926,15 @@ namespace eosio {
          if (length > 0) {
             std::string organization{buf, (size_t)length};
             if (is_string_valid_name(organization)){
-               account_name participant{organization};
-               
+               participant_name_ = account_name{organization};
+
                std::lock_guard<std::shared_mutex> connection_guard(my_impl->connections_mtx);
-               if(my_impl->security_group.is_in_security_group(participant)) {
-                  participant_name_ = participant;
-                  return true;
+               if(my_impl->security_group.is_in_security_group(*participant_name_)) {
```

Comment:
using the verified temporary (has no use the rest of the life of the connection) flag is complex, instead move line 931 through 934 to the start_session logic, it will be much cleaner than caching the value here, and if a new security_group came in, between verify_certificate and the processing in start_session, we would be setting the wrong set_participating value.